core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
In DR's IR, an undecoded instr (OP_UNDECODED) is either Level 0 (could be a bundle of instrs) or Level 1 (just 1 instr) and has not been decoded yet other than for length.  If it has been decoded, it has an opcode or it's invalid (OP_INVALID) according to our decoder.  You're trying to use OP_UNDECODED for a new category of things which the rest of the code was not designed for.  Asking for an opcode is designed to do an auto-magic up-decode to Level 2.

I think it would be better to stick with what existing code is already using: use OP_INVALID for an instr that was decoded to Level 2 but whose opcode is not known to our decoder: i.e., we think it's an invalid encoding.  Do not use OP_UNDECODED for that case.